civismlext/test/test_preprocessing.py

```diff
@@ -732,3 +746,47 @@ def test_transform_no_level_overlap():
         msg = "No overlap between levels in column 'age' and " + \
               "levels seen during fit"
         assert msg in w[0].message.args[0]
+
+
+def test_expand_all_na(data_raw):
+    # output df should only have nan columns for columns which
+    # actually had nans during fit
+    df_expected = pd.concat([
+        pd.Series([0., 1., 0.], dtype='float32', name='djinn_type_effrit'),
+        pd.Series([1., 0., 0.], dtype='float32', name='djinn_type_marid'),
+        pd.Series([0., 0., 1.], dtype='float32', name='djinn_type_sila'),
+        pd.Series([1., np.nan, 3.], dtype='float32', name='fruits'),
+        pd.Series([0., 1., 0.], dtype='float32', name='fruits_NaN'),
+        pd.Series([1., 0., 0.], dtype='float32', name='animal_cat'),
+        pd.Series([0., 1., 0.], dtype='float32', name='animal_dog'),
+        pd.Series([0., 0., 1.], dtype='float32', name='animal_NaN'),
+    ], axis=1)
+    expander = DataFrameETL(cols_to_expand=['djinn_type', 'animal'],
+                            cols_to_drop=['pid', 'age'],
+                            dummy_na='all',
+                            dataframe_output=True)
+    expander.fit(data_raw)
+    assert expander.unexpanded_nans == {'fruits': True}
+
+    df_out = expander.transform(data_raw)
+    assert df_out.equals(df_expected)
+
+
+@pytest.mark.skipif(sys.version_info < (3,), reason='requires python 3')
+def test_dummy_na_true_deprecated(data_raw):
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        DataFrameETL(cols_to_drop=['pid'],
+                     cols_to_expand=['djinn_type', 'fruits', 'animal'],
+                     dummy_na=True)
+        assert len(w) == 1
+        assert issubclass(w[-1].category, DeprecationWarning)
+
+
+def test_dummy_na_bad_value(data_raw):
+    with pytest.raises(ValueError):
```

Comment:
It's worth checking that the `ValueError` is the expected one. Otherwise this might be hiding an unrelated `ValueError`.